packages/portfolio-contract/src/resolver/resolver.exo.ts

```diff
@@ -22,36 +21,46 @@ import { E } from '@endo/far';
 import { M } from '@endo/patterns';
 import { TxStatus, TxType } from './constants.js';
 import type {
-  CCTPSettlementArgs,
-  CCTPSettlementOfferArgs,
-  CCTPTransactionKey,
   PublishedTx,
+  TransactionSettlementOfferArgs,
+  TxId,
+} from './types.js';
+import {
+  ResolverOfferArgsShapes,
+  TransactionSettlementOfferArgsShape,
 } from './types.js';
-import { CCTPSettlementArgsShape, ResolverOfferArgsShapes } from './types.js';
 
-type CCTPTransactionEntry = {
+type TransactionEntry = {
   destinationAddress: AccountId;
   amountValue: bigint;
   vowKit: VowKit<void>;
+  type: TxType;
 };
 
 const trace = makeTracer('Resolver');
 
-const ClientFacetI = M.interface('CCTPResolverClient', {
-  registerCCTPTransaction: M.call(M.string(), M.nat()).returns(VowShape),
+const ClientFacetI = M.interface('ResolverClient', {
+  registerTransaction: M.call(
+    M.or(...Object.values(TxType)),
+    M.string(),
+    M.nat(),
+  ).returns(VowShape),
 });
 
 const ReporterI = M.interface('Reporter', {
-  insertPendingTransaction: M.call(M.bigint(), M.string()).returns(),
-  completePendingTransaction: M.call(M.string(), M.string()).returns(),
+  insertPendingTransaction: M.call(M.string(), M.string(), M.nat()).returns(),
+  completePendingTransaction: M.call(
+    M.string(),
+    M.or(TxStatus.SUCCESS, TxStatus.FAILED),
+  ).returns(),
 });
 
-const ServiceFacetI = M.interface('CCTPResolverService', {
-  settleCCTPTransaction: M.call(CCTPSettlementArgsShape).returns(),
+const ServiceFacetI = M.interface('ResolverService', {
+  settleTransaction: M.call(TransactionSettlementOfferArgsShape).returns(),
 });
 
 const InvitationMakersFacetI = M.interface('ResolverInvitationMakers', {
-  SettleCCTPTransaction: M.callWhen().returns(M.remotable()),
+  SettleTransaction: M.callWhen().returns(M.remotable()),
```

Comment:
```suggestion
  SettleTransaction: M.callWhen().returns(M.remotable('Invitation')),
```

or:

```suggestion
  SettleTransaction: M.callWhen().returns(InvitationShape),
```

not critical, but sometimes handy for debugging